src/components/ChatEntry.js

```diff
@@ -1,22 +1,43 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
-const ChatEntry = (props) => {
+
+const ChatEntry = ({id, sender, body, timeStamp, liked, handleLikes}) => {
+  
+  const updateMessage = ()=> {
+    handleLikes({
+      id,
+      sender,
+      body,
+      timeStamp,
+      liked :!liked
+    })
+  }
+  
+  const heartColor = liked ? '❤️': '🤍' ;
+  const position = id %2 ? 'local' : 'remote';
   return (
-    <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+    <div className={`chat-entry ${position}`}>
+      <h2 className="entry-name">{sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p>{body}</p>
+        <p className="entry-time"><TimeStamp time={timeStamp} /></p>
+        <button onClick={updateMessage} className="like" >{heartColor}</button>
+        
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id:PropTypes.number,
```

Comment:
I'd recommend making `id` required here since it's needed for the like button to work.